orc8r/gateway/python/magma/common/service.py

```diff
@@ -295,28 +297,47 @@ def _set_grpc_poll_strategy(self):
         """
         os.environ['GRPC_POLL_STRATEGY'] = 'epoll1,poll'
 
-    def _setup_logging(self):
-        """
-        Setup the logging for the service
-        """
+    def _get_log_level_from_config(self) -> Optional[int]:
         if self._config is None:
-            config_level = None
-        else:
-            config_level = self._config.get('log_level', None)
-            if config_level is None and self._mconfig is not None:
-                config_level = LogLevel.Name(self._mconfig.log_level)
+            return None
+        log_level = self._config.get('log_level', None)
+        if log_level is None:
+            return None
+        # convert from log level string to LogLevel enum value
         try:
-            proto_level = LogLevel.Value(config_level)
+            proto_level = LogLevel.Value(log_level)
         except ValueError:
             logging.error(
-                'Unknown logging level in config: %s, defaulting to INFO',
-                config_level,
+                'Unknown logging level in config: %s, ignoring',
+                log_level,
+            )
+            return None
+        return proto_level
+
+    def _get_log_level_from_mconfig(self) -> Optional[int]:
+        if self._mconfig is None:
+            return None
+        return self._mconfig.log_level
+
+    def _setup_logging(self):
+        """Set up log level from config values"""
```

Comment:
Edit this docstring so it specifies that the yml is preferred over mconfig. But long term this is confusing that we have multiple places to specify the log level

Reply:
Will edit. I think the intention is to always be able to override configs from the AGW locally, which is super useful for AGW-centric people. (But maybe it is counterintuitive? not sure)